gax-httpjson/src/main/java/com/google/api/gax/httpjson/ManagedHttpJsonChannel.java

```diff
@@ -0,0 +1,175 @@
+import com.google.api.client.http.HttpRequestFactory;
+import com.google.api.client.http.HttpResponse;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.UrlEncodedContent;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.JsonObjectParser;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.util.GenericData;
+import com.google.api.core.ApiFuture;
+import com.google.api.core.BetaApi;
+import com.google.api.core.SettableApiFuture;
+import com.google.api.gax.core.BackgroundResource;
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
+
+@BetaApi
+public class ManagedHttpJsonChannel implements HttpJsonChannel, BackgroundResource {
+  private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+
+  private final Executor executor;
+  private final HttpTransport httpTransport;
+  private final String endpoint;
+  private final JsonFactory jsonFactory;
+  private final ImmutableList<HttpJsonHeaderEnhancer> headerEnhancers;
+
+  private ManagedHttpJsonChannel(
+      Executor executor,
+      HttpTransport httpTransport,
+      String endpoint,
+      JsonFactory jsonFactory,
+      List<HttpJsonHeaderEnhancer> headerEnhancers) {
+    this.executor = executor;
+    this.httpTransport = httpTransport;
+    this.endpoint = endpoint;
+    this.jsonFactory = jsonFactory;
+    this.headerEnhancers = ImmutableList.copyOf(headerEnhancers);
+  }
+
+  public <ResponseT, RequestT> ApiFuture<ResponseT> issueFutureUnaryCall(
+      HttpJsonCallOptions callOptions, RequestT request) {
+
+    final SettableApiFuture responseFuture = SettableApiFuture.create();
+
+    executor.execute(
+        new Runnable() {
+          @Override
+          public void run() {
+            try {
+              GenericData tokenRequest = new GenericData();
+              // TODO convert request to GenericData
+
+              UrlEncodedContent content = new UrlEncodedContent(tokenRequest);
+              HttpRequestFactory requestFactory = httpTransport.createRequestFactory();
+              HttpRequest request =
+                  requestFactory.buildPostRequest(new GenericUrl(endpoint), content);
+              for (HttpJsonHeaderEnhancer enhancer : headerEnhancers) {
+                enhancer.enhance(request.getHeaders());
+              }
+              request.setParser(new JsonObjectParser(jsonFactory));
+              HttpResponse response = request.execute();
+
+              GenericData responseData = response.parseAs(GenericData.class);
+              // TODO convert GenericData to response type
+
+              responseFuture.set(responseData);
+
+            } catch (Exception e) {
+              responseFuture.setException(e);
+            }
+          }
+        });
+
+    return responseFuture;
+  }
+
+  // TODO implement the following
```

Comment:
I believe it should be implemented now. Looks straightforward, as, probably the only thing that should be done is cancelling the background task appropriately in shutdown methods, and checking its status in getter methods.